packages/logic/src/services/ValidationService/Validators/CompareAgePnrValue.spec.ts

```diff
@@ -0,0 +1,26 @@
+import { compareAgePnrValue } from "./CompareAgePnrValue";
+
+const element = document.createElement("input");
+
+it.each`
```

Comment:
Egentligen tycker jag det blir lite smårörigt med `it.each(..)` och speciellt här där det är så många parametrar.

Jämför med att ha separata `it(..)`:

```ts
it("should handle null and undefined", () => {
    const config = { pnrValueShouldBeOlder: true };
    expect(compareAgePnrValue.validation(undefined, element, { otherField: "1212111111", ...config })).toBeTruthy();
    expect(compareAgePnrValue.validation(null, element, { otherField: "1212111111", ...config })).toBeTruthy();
    expect(compareAgePnrValue.validation("1212111111", element, { otherField: null, ...config })).toBeTruthy();
    expect(compareAgePnrValue.validation("1212111111", element, { otherField: undefined, ...config })).toBeTruthy();
});

it("should handle invalid values", () => {
    const config = { pnrValueShouldBeOlder: true };
    expect(compareAgePnrValue.validation("abcd1234", element, { otherField: "1212111111", ...config })).toBeTruthy();
    expect(compareAgePnrValue.validation("1212111111", element, { otherField: "abcd1234", ...config })).toBeTruthy();
});

it("should return true when age is valid", () => {
    /* ... */
});

it("should return false when age is invalid", () => {
    /* ... */
});

it("should return ??? when age is the same", () => {
    /* ... */
});
```

Jag kan köpa `it.each(..)` när det bara handlar om att variera en eller maximalt två variabler men när det mest blir boilerplate där tom testets beskrivning ligger med i `it.each(..)` samt att själva testfallet inte ens innehåller någon repetition då tycker jag bara det blir drygt att läsa och förstå vad som faktiskt testas. Både när jag läser koden men också när jag läser test-resultatet.

> should return true when value="1301012397", otherField="1313131313", and pnrValueShouldBeOlder=true because value is older than otherField and older values are allowed

Det blir så väldigt pratigt, jämför med:

> should return true when value is older and older values are allowed

eller som jag föreslår ovan:

> should return true when value is valid

där man testar båda varianterna (yngre med true, äldre med false).

Reply:
Fixat.